src/main/java/cubicchunks/client/ClientEventHandler.java

```diff
@@ -94,48 +94,35 @@ public void initGuiEvent(InitGuiEvent.Post event) {
             GuiVideoSettings gvs = (GuiVideoSettings) currentGui;
             try {
                 GuiOptionsRowList gowl = (GuiOptionsRowList) gvs.optionsRowList;
-                GuiOptionsRowList.Row row = this.createRow(100, CubicChunks.Config.IntOptions.VERTICAL_CUBE_LOAD_DISTANCE, gvs.width);
+                GuiOptionsRowList.Row row = this.createRow(100, CubicChunks.Config.IntOptions.RENDER_CHUNK_SIZE_BIT, CubicChunks.Config.IntOptions.VERTICAL_CUBE_LOAD_DISTANCE, gvs.width);
                 gowl.options.add(1, row);
             } catch (NoSuchFieldError err) {
                 int idx = 3;
                 int btnSpacing = 20;
                 CubicChunks.LOGGER.error("Couldn't add vertical view distance options, maybe optifine is installed? Attempting optifine-specific "
                         + "option add", err.toString());
-                gvs.buttonList.add(idx, new GuiCustomSlider(100, gvs.width / 2 - 155 + 160, gvs.height / 6 + btnSpacing * (idx / 2) - 12,
+                gvs.buttonList.add(idx++, new GuiCustomSlider(100, gvs.width / 2 - 155 + 160, gvs.height / 6 + btnSpacing * (idx / 2) - 12,
                         CubicChunks.Config.IntOptions.VERTICAL_CUBE_LOAD_DISTANCE));
-                // reposition all buttons except the last 4 (last 3 and done)
-                for (int i = 0; i < gvs.buttonList.size() - 4; i++) {
+                gvs.buttonList.add(idx, new GuiCustomSlider(101, gvs.width / 2 - 155 + 160, gvs.height / 6 + btnSpacing * (idx / 2) - 12,
+                        CubicChunks.Config.IntOptions.RENDER_CHUNK_SIZE_BIT));
+                // reposition all buttons except "done".
+                for (int i = 0; i < gvs.buttonList.size() - 1; i++) {
                     GuiButton btn = gvs.buttonList.get(i);
                     int x = gvs.width / 2 - 155 + i % 2 * 160;
                     int y = gvs.height / 6 + 21 * (i / 2) - 12;
                     btn.x = x;
                     btn.y = y;
                 }
-                // now position the last 3 buttons excluding "done" to be 3-in-a-row
-                for (int i = gvs.buttonList.size() - 4; i < gvs.buttonList.size() - 1; i++) {
-                    GuiButton btn = gvs.buttonList.get(i);
-
-                    int newBtnWidth = 150 * 2 / 3;
-                    int minX = gvs.width / 2 - 155;
-                    int maxX = gvs.width / 2 - 155 + 160 + btn.width;
-
-                    int minXCenter = minX + newBtnWidth / 2;
-                    int maxXCenter = maxX - newBtnWidth / 2;
-
-                    int x = minXCenter + (i % 3) * (maxXCenter - minXCenter) / 2 - newBtnWidth / 2;
-                    int y = gvs.height / 6 + 21 * (gvs.buttonList.size() - 4) / 2 - 12;
-
-                    btn.x = x;
-                    btn.y = y;
-                    btn.width = newBtnWidth;
-                }
+                GuiButton btn = gvs.buttonList.get(gvs.buttonList.size() - 1);
+                btn.y += 21;
```

Comment:
What does the optifine GUI look like after your changes?

Reply:
Thats how:
![screenshot from 2018-01-06 18-05-05](https://user-images.githubusercontent.com/6910858/34640877-29b90368-f30c-11e7-8081-a35d7f6f3ce6.png)

Reply:
Does it work with the minimum window size that doesn't change the scaling?

Reply:
No. "done" button is hiding. Want to switch to 3 buttons in a row or just reduce button height?

Reply:
Try to make the 6 bottom buttons be 3 buttons in a row